version: '3.8'

services:
  postgres:
    image: postgres:latest
    environment:
      POSTGRES_USER: app
      POSTGRES_PASSWORD: app
      POSTGRES_DB: app
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app-network

  app:
    build:
      context: ./app
    volumes:
      - ./app:/app
      - static_volume:/app/static
    ports:
      - "8000:8000"
    depends_on:
      - postgres
    environment:
      DEBUG: "False"
      DB_NAME: "app"
      DB_USER: "app"
      DB_PASSWORD: "app"
      DB_HOST: "postgres"
      DB_PORT: 5432
    restart: on-failure:2
    networks:
      - app-network

  nginx:
    image: nginx:latest
    volumes:
      - ./conf/nginx.conf:/etc/nginx/nginx.conf
      - static_volume:/static
    ports:
      - "80:80"
    depends_on:
      - app
      - postgres
    networks:
      - app-network

volumes:
  postgres_data:
  static_volume:

networks:
  app-network:
    driver: bridge
